include/clang/AST/Decl.h

```diff
@@ -1120,6 +1129,19 @@ class VarDecl : public DeclaratorDecl, public Redeclarable<VarDecl> {
     return false;
   }
 
+  /// \brief - Return true if this variable has bounds declared for it.
+  bool hasBoundsExpr() const;
```

Comment:
What's your thoughts on making these accessors public?  They seem useful to have around.  At minimum, I'd expect them to be at least as accessible as the underlying BoundsExpr pointer itself (protected).